lib/pure/math.nim

```diff
@@ -981,27 +992,48 @@ func euclMod*[T: SomeNumber](x, y: T): T {.since: (1, 5, 1).} =
   if result < 0:
     result += abs(y)
 
-when not defined(js):
-  func c_frexp*(x: float32, exponent: var int32): float32 {.
-      importc: "frexp", header: "<math.h>".}
-  func c_frexp*(x: float64, exponent: var int32): float64 {.
-      importc: "frexp", header: "<math.h>".}
-  func frexp*[T, U](x: T, exponent: var U): T =
-    ## Split a number into mantissa and exponent.
-    ##
-    ## ``frexp`` calculates the mantissa m (a float greater than or equal to 0.5
-    ## and less than 1) and the integer value n such that ``x`` (the original
-    ## float value) equals ``m * 2**n``. frexp stores n in `exponent` and returns
-    ## m.
-    ##
-    runnableExamples:
-      var x: int
-      doAssert frexp(5.0, x) == 0.625
-      doAssert x == 3
-    var exp: int32
-    result = c_frexp(x, exp)
-    exponent = exp
+func frexp[T: float32|float64](x: T): tuple[frac: T, exp: int] {.inline.} =
+  ## Splits a number into mantissa and exponent.
+  ## The mathematical formula for this method is: number = frac * 2 ^ exp.
```

Comment:
```suggestion
  ## The mathematical formula for this method is: number = frac * 2 ^ exp.
  runnableExamples: doAssert frexp(8.0) == (0.5, 4)
```
(yes, it'll be run and shown in docs thanks to export frexp)

Reply:
It seems to have some bugs with `frexp` for special value handling(0.0, -0.0, NaN, -Inf, Inf) in my win10 machine.

Reply:
I even write a `test.c` program, it seems there is something wrong with `frexp` c99.

Reply:
=> https://github.com/nim-lang/Nim/issues/17017#issuecomment-777932990